### DISCUSSION TITLE
Connecting the Dualshock 2 to the Tang Nano 20k

### DISCUSSION THREAD

**Discussion by otaviocmaciel** [nand2mario/nestang#26] [General]
Hello dear friends! I didn't find the wiring diagram to connect the Dualshock 2 to the Tang Nano 20k, maybe I didn't search correctly. I will describe it to make it easier for others (for those who don't have the adapter).

If you open GPRJ FPGA project you can see this (in the FloorPanel):

![image](https://github.com/nand2mario/nestang/assets/93693421/2728bd0f-6204-4770-89a5-63a45f33f480)

By this way, the wiring diagram to conect the DS2 to the Tang Nano 20k is:

![image](https://github.com/nand2mario/nestang/assets/93693421/ba1d7ee8-23e7-4083-9c8c-d8a0a80eb6a2)

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/tavin/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/tavin/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

  | TN20k (Controller 1) | TN20k   (Controller 2) | DS2 (Color)
-- | -- | -- | --
VCC | 3V3 | 3V3 | Red
GND | GND | GND | Black
CLK | 17 | 52 | Blue
CS | 18 | 72 | Yellow
MISO | 19 | 71 | Brown
MOSI | 20 | 53 | Orange

</body>

</html>

![20230908_001435.jpg](https://github.com/nand2mario/nestang/assets/93693421/bf1ca4f6-6f9d-4c88-9d0c-6139db83d829)

Ready! Everything should work.

**Comment by Chandler-Kluser** [comment]
I believe this kind of information is not well indicated in documentation, you can make a PR to add a markdown guide for newbies.

The thing that actually is the most close of your request is [this wiring guide for the Tang Nano 9k version (with blueretro adapter)](https://github.com/Chandler-Kluser/nestang9k-ps2/blob/main/docs/README.md)

**Comment by Chandler-Kluser** [comment]
I have done a 5-minute search and I have found [this guide](https://github.com/nand2mario/nestang/wiki/Connecting-the-Sipeed-PS2-Adapters) from this Official repository in the wiki page.

**Reply by otaviocmaciel** [reply]
Ok, but where is the cable color indication and where should it be connected? I don't have an adapter.

**Reply by Chandler-Kluser** [reply]
There is also a video on youtube with the color wiring, in the link I fixed the timestamp for you:
https://www.youtube.com/watch?v=UNcErsqld5s&t=506s
You can also check [this picture](http://www.billporter.info/wp-content/uploads/2010/05/wiring.jpg):
![](http://www.billporter.info/wp-content/uploads/2010/05/wiring.jpg)

**Reply by otaviocmaciel** [reply]
Ok! Thank you👍🏻
